Detectors/Upgrades/ALICE3/Passive/src/Pipe.cxx

```diff
@@ -92,35 +95,69 @@ void Alice3Pipe::ConstructGeometry()
     LOG(FATAL) << "Could not find the top volume";
   }
 
+  //---------------- Outermost Be pipe around the IP ----------
+  // Outer pipe has to be filled with vacuum. There we have also TRK layers, which we don't want to depend on the pipe volume.
+  // Eventually, we will depend on some information passed from the outside.
+  // For A3PIP-only simulations, we don't want TRK's shade.
+  // Strategy used here is to use a composite shape where shapes of TRK layers are subtracted to the vacuum volume
+  TGeoTube* outerBeTube = new TGeoTube("OUT_PIPEsh", mBeOuterPipeRmin, mBeOuterPipeRmin + mBeOuterPipeThick, mOuterIpHLength);
+  TGeoVolume* outerBeTubeVolume = new TGeoVolume("OUT_PIPE", outerBeTube, kMedBe);
+  outerBeTubeVolume->SetLineColor(kBlue);
+
+  TGeoTube* outerBerylliumTubeVacuumBase = new TGeoTube("OUT_PIPEVACUUM_BASEsh", mBeInnerPipeRmin + mBeInnerPipeThick, mBeOuterPipeRmin, mOuterIpHLength); // Vacuum filling for outer pipe
+  TGeoCompositeShape* outerBerylliumTubeVacuumComposite;                                                                                                   // Composite volume to subctract to vacuum
+  TGeoVolume* outerBerylliumTubeVacuumVolume;                                                                                                              // Final volume to be used
+
+  TString compositeFormula{"OUT_PIPEVACUUM_BASEsh"}; // If pipe is alone we won't subctract anything
+  TString subtractorsFormula;
+
+  if (!mIsAlone) {
+    std::vector<TGeoTube*> trkLayerShapes;
+    std::vector<std::array<float, 3>> layersQuotas = {std::array<float, 3>{0.5f, 15.f, 50.e-4}, // TODO: Set layers dynamically. {radius, zLen, thickness}
```

Comment:
I don't have enough insight in the geometry organisation, but wouldn't it be more natural to provide a handle such that the tracker can remove the relevant parts from the beam pipe vacuum when it gets added? That would also avoid the complication from knowing here how some other part of the detector is implemented. But maybe I am not seeing the full picture.

Reply:
The beam pipe is filled with vacuum rather than air, hence it is not possible to append the inner part of the TRK directly to the barrel/cave as it will create a volume overlap as the vacuum medium and the silicon layers will be at the same level of the geometry topology, creating possible inconsistencies during particle propagation.
In a possibile future where the TRK description will be more complex, I have in mind some options to tackle this design choice, which describes also a kind of unprecedented layout (to me, at least).

To address specific points:
> a handle such that the tracker can remove the relevant parts from the beam pipe vacuum when it gets added?

Shapes can be subctracted each others, then the volume is instantiated out of their "combination". I am not an experienced geometry user but for what I learned it is not possible to "excavate" a volume using another one once it is created.
Moreover, concerning how geometry is used in O2 simulations, to make detector objects inter-operate is not something that is foreseen/common practice.
To get detectors/passives back from FairRoot and apply modifications might not be so natural as each detector activation/creation/forwarding is done in separate ways: each item is kind of _agnostic_ of the fairroot interface and of the other objects.
 
> That would also avoid the complication from knowing here how some other part of the detector is implemented

This wil avoid duplication of the description, which is indeed the line we want to follow as much as possible, especially if the complexity of the detector layout is going to increase a lot.
In terms of complication of code structure it would however make Pipe to depend from TRK such that either of them will need to be aware of the description of the other in order to know what to remove which is very similar to current layout. 

For the moment I would leave this implementation as it is, as it describes a reasonably simple layout. 
After the major cleanup that the TRK is undergoing, it will be easier to implement some hybrid solution such as appending the three innermost TRK layer volumes to the VACUUM volumes of the beampipe if present (they can be accessed into the TGeoManager interface by name and react accordingly if not present (e.g. TRK-only simulation)).

Is it feasible for you?

Reply:
@mconcas: Fine with me for now, but maybe we can still discuss (at some convenient occasion, not urgent) how to handle this in the future, also in view of comparison of different inner tracker geometries.